docker.go

```diff
@@ -57,6 +67,21 @@ func runDockerPin(cmd *cobra.Command, args []string) error {
 	return ioutil.WriteFile(ifDash(*dockerfile, "/dev/stdout"), n, 0644)
 }
 
+func runDockerBasePin(cmd *cobra.Command, args []string) error {
+	baseImageAndVersion := args[0]
+	baseImage := strings.Split(baseImageAndVersion, ":")[0]
+	dockerClient, err := docker.NewClientWithOpts(docker.FromEnv)
+	if err != nil {
+		return err
+	}
+	di, diErr := dockerClient.DistributionInspect(cmd.Context(), baseImageAndVersion, "")
```

Comment:
You can redefine variables in Go (with the same type), just call `diErr` `err` and that'll work.